### ISSUE TITLE
Add Code of Conduct

### ISSUE THREAD

**Issue by erri120** [Nexus-Mods/NexusMods.App#275]
Why: [Open Source Guide](https://opensource.guide/code-of-conduct/), [GitHub Docs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project)
How (templates):
- [Contributor Covenant](https://www.contributor-covenant.org/)
- [Citizen Code of Conduct](https://web.archive.org/web/20200330154000/http://citizencodeofconduct.org/)

**Comment by Continous** [comment]
I would really advise against the contributor covenant. There's a lot of justified and unjustified controversy surrounding it. I generally find code of conducts unnecessary beyond a very simple, "Please act appropriately." One does not need to be told how to act, and we all know there will be a clause in the code of conduct that makes the whole thing null and void by saying it is non-exhaustive. Just because the open source guide and github suggest it's something that is necessary for a project doesn't necessarily make it the case. Vortex hasn't had a CoC and hasn't had any significant controversy.

**Comment by halgari** [comment]
> I would really advise against the contributor covenant. There's a lot of justified and unjustified controversy surrounding it.

Okay @Continous can you link said information? First step in any technical discussion is gathering information.

**Comment by Continous** [comment]
> > I would really advise against the contributor covenant. There's a lot of justified and unjustified controversy surrounding it.
> 
> Okay @Continous can you link said information? First step in any technical discussion is gathering information.

While it's not my place to suggest what is right or wrong on the manner, much less what people find appropriate, I just feel it's better to sidestep the issue entirely if possible. I'd much prefer my favorite website and it's apps be as controversy free as possible. So I'll limit my personal opinions on the matter.

With that said, through a simple google search one can find a plethora of drama surrounding the code of conduct, and honestly code of conducts in general, especially after its adoption by the Linux kernel.

ItsFOSS had a good write up of the Linux controversy.
https://itsfoss.com/linux-code-of-conduct/

Ruby had their own little melt down over it.
https://bugs.ruby-lang.org/issues/12004

OpenStreetMap had a similar, but more civil, discussion on the matter.
https://github.com/openstreetmap/openstreetmap-website/issues/1299

There's non-insignificant issues surrounding the contributor covenant, and specifically it's author, and I have seen at least one instance where a Code of Conduct has been used not to ensure proper conduct within a project, but to ideologically isolate and expel people from a project. The Contributor Covenant and its look-a-likes are uniquely predisposed to this misuse in my opinion since they inherently include charged language. Specifically:

Listing a ton of characteristics for which people should be protected from harassment is unnecessary and linguistically implies this to be an exhaustive set. Where the Contributor Covenant pledges to, "make participation in our community a harassment-free experience for everyone, regardless of..." it simply opens itself up to obvious controversy. Such phrasing only leaves questions. Simply stating that we pledge to make participation in our community a harassments-free experience for everyone, would already cover all of these situations.

A code of conduct, in my opinion, if necessary at all should be custom written by the administrators or caretakers of the project, with the specific conduct they'd like to see in mind. For the most part, if something that is unacceptable occurs, it will not matter if it was in the code of conduct, it is unacceptable and should not have occurred, so outline what is and is not acceptable is obviously an exercise in futility. Furthermore, what is and is not acceptable differs depending on the project and community. This is not something that is fixed in stone either.

Again though, I digress. I don't particularly care and doubt I'd be particularly effected by the addition of any CoC as I am not personally planning on doing anything that'd violate any conceivably reasonable CoC. I just figured it'd be a good idea to provide the context surrounding the other adoptions of similar code of conducts to far more professional and popular projects.

**Comment by Aragas** [comment]
That's an interesting topic. It seems that a CoC could be directly used for abuse, like with the [Opal](https://opalrb.com/) project. While this seems to be more of an exception, this does look concerning.
The modding community as a whole contains people of different views and beliefs, I'd say it's one of the most diversified communities. NexusMods as a company did a good job at moderating the community, preventing controversies most of the time.
If a CoC is necessary, I would suggest to create a custom one, based _only_ on existing NexusMods policies to keep the expectations of the moderation consistent.
Otherwise, the best option would be to keep it stupid simple and just let the core maintainers (NexusMods) to moderate as they seem fit. It still will be based on NexusMods policies anyway.

**Comment by halgari** [comment]
Yes, I think the distinction needs to be made here that while this app is open source and GPL3, the project governance is centralized and backed by a corporate entity. So the question of "how do we deal with trouble makers" is pretty simple: the same way we deal with troublemakers on the site itself. 

That being said, we should probably hammer out some doc that says "interactions on this Github Project are expected to conform to the same standards of conduct as the nexusmods.com site". At least in that way the relationship between the project and the site maintainers is made explicit.

**Comment by Continous** [comment]
On this I think everyone can agree.
